pandas_transform/pandas_transform.md

```diff
@@ -0,0 +1,995 @@
+covid_testing = covid_testing.replace(["positive","negative"],[1,0])
+
+print(covid_testing.loc[:,["first_name", "last_name", "result"]])
+
+</script>
+</lia-keep>
+</div>
+
+<div class= "warning">
+The order and length of these lists matter! Each element in the first list will be replaced by the element in the **same position** in the second list.
+</div>
+
+### Quiz: Transforming DataFrames
+
+@sage
+<div class="python_data_init">
+<lia-keep>
+<script type="text/x-sage">
+import pandas as pd
+covid_testing = pd.read_csv('https://raw.githubusercontent.com/arcus/education_modules/embedded_code/a_sample_module_template/covid_testing.csv')
+</script>
+</lia-keep>
+</div>
+
+You can change the all of the entries in a column to uppercase using the method `.str.upper()`. The code below **prints** the `last_name` column  of `covid_testing`  in uppercase:
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+print(covid_testing.loc[:, "last_name"].str.upper())
+</script>
+</lia-keep>
+</div>
+<br>
+
+How would you **change** the the `last_name` column of `covid_testing` to be uppercase?
+
+[( )] `covid_testing.loc[:, "last_name"].str.upper()`
+[( )] `covid_testing.loc[:, "last_name"] == covid_testing.loc[:, "last_name"].str.upper()`
+[(X)] `covid_testing.loc[:, "last_name"] = covid_testing.loc[:, "last_name"].str.upper()`
+***
+<div class = "answer">
+- `covid_testing.loc[:, "last_name"].str.upper()` This command will change the way the `last_name` column is **displayed**, but will not make any changes to the DataFrame itself.
+
+- `covid_testing.loc[:, "last_name"] == covid_testing.loc[:, "last_name"].str.upper()` The double equals `==` mean that this is a **condition** testing whether or not the entry int the `last_name` column is already uppercase.
+
+- `covid_testing.loc[:, "last_name"] = covid_testing.loc[:, "last_name"].str.upper()` The single equals `=` **assigns** the entry in the `last_name` column to the uppercase version.
+</div>
+***
+
+<div class = "learnmore">
+
+The method `.str.lower()` will make every entry in a column lower case. There are several [other ways to manipulate the presentation of text](https://pandas.pydata.org/docs/reference/api/pandas.Series.str.lower.html) in a pandas series.
+
+</div>
+
+## Additional Resources
+
+The creators of the pandas package have [great tutorials](https://pandas.pydata.org/docs/getting_started/index.html) with very thorough examples.
+
+
+## Feedback
+
+In the beginning, we stated some goals.
+
+**Learning Objectives:**
+
+@learning_objectives
+
+We ask you to fill out a brief (5 minutes or less) survey to let us know:
+
+* If we achieved the learning objectives
+* If the module difficulty was appropriate
+* If we gave you the experience you expected
+
+We gather this information in order to iteratively improve our work. Thank you in advance for filling out [our brief survey](https://redcap.chop.edu/surveys/?s=KHTXCXJJ93&module_name=%22Transform+Data+with+Pandas%22)!
```

Comment:
The module name here doesn't match the current title for the module (although actually I think I prefer this one!)

Reply:
Good call, I'll make this the actual module name!